rfd/0097-teleport-connect-usage-metrics.md

```diff
@@ -80,142 +110,70 @@ changes in a given period of time.
 
 ## Events
 
-### `connect.start`
-
-Starting the app.
-
-Event properties:
-
-- `os`: one of `mac`/`windows`/`linux`
-- `connect_version`: string
-
-### `connect.cluster.add`
-
-Successful adding a cluster.
-
-Event properties:
-
-- `root_cluster_name`: string - name of the added cluster
-
 ### `connect.cluster.login`
 
 Successful login to a cluster.
 
 Event properties:
 
-- `root_cluster_name`: string
+- `cluster_name`: string (anonymized)
+- `user_name`: string (anonymized)
 - `connector_type`: string
+- `os`: string (set once on a user properties)
+- `arch`: string (set once on a user properties) - CPU architecture
+- `os_version`: string (set on a user properties)
+- `connect_version`: string (set on a user properties)
+- `distinct_id`: string
 
-### `connect.workspace.change`
-
-Changing a workspace.
-
-Event properties:
-
-- `target_root_cluster_name`: string
-
-### `connect.leafClusterPicker.change`
-
-Changing a leaf cluster.
-
-Event properties:
-
-- `root_cluster_name`: string
-- `target_leaf_cluster_name`: string
-
-### `connect.tab.open`
-
-Opening a tab.
-
-Event properties:
-
-- `root_cluster_name`: string
-- `leaf_cluster_name`: string (optional)
-- `tab_kind`: one
-  of `resource_list`/`terminal_ssh`/`terminal_kube`/`terminal_shell`/`db_proxy`/`create_access_request`/`review_access_request`
-
-### `connect.connectionsPicker.item.select`
+### `connect.protcol.run`
 
-Selecting an item from the connections picker.
+Connecting to the protocol.
 
 Event properties:
 
-- `root_cluster_name`: string
-- `leaf_cluster_name`: string (optional)
-- `connection_kind`: one of `ssh`/`kube`/`db`
-
-### `connect.commandBar.command.run`
-
-Running a command from command bar.
-
-Event properties:
-
-- `root_cluster_name`: string
-- `leaf_cluster_name`: string (optional)
-
-### `connect.commandBbar.autocomplete.select`
-
-Selecting autocomplete item from the command bar suggestions.
-
-Event properties:
-
-- `root_cluster_name`: string
-- `leaf_cluster_name`: string
-- `suggestion`: one of `ssh_login`/`ssh_server`/`database`
-
-### `connect.resourceTable.search`
-
-Performing a resource search.
-
-Event properties:
-
-- `root_cluster_name`: string
-- `leaf_cluster_name`: string
-- `resource_kind`: one of `ssh`/`kube`/`db`
-- `is_advanced`: boolean
+- `cluster_name`: string (anonymized)
+- `user_name`: string (anonymized)
+- `protocol`: one of `ssh`/`proxy_db`/`kube`
```

Comment:
Just to be clear, for the db proxies we're going to count the number of times the user has set up a proxy, not each connection, right?

We can do both, it's just a matter of deciding which one we want to track.

Reply:
I would collect an event for the `proxy_db` protocol when the user has set up a proxy.
Counting actual connections to the proxy can be done using another event like `db_connection` in the future.